### PR TITLE
integrate over non-mesh dimension

### LINKED ISSUE #190: integrate process only works on mesh dim which may be different from the analysis dim
The example here is a 1d beam/truss element in 3d space. We want to integrate over 1D elements. Currently I work around this issue by replicating the `Integrator::process(Mesh* m)` code locally and process each element.

To reduce code duplication I want to propose the following change in core, which leaves the current api in tact, but adds a default argument to the process call.

e.g.
 
`Integrator::process(Mesh* m, int dim=-1)`

if the dim is `-1` the current behavior is maintained, otherwise we integrate over the user supplied dimension.

@cwsmith @wrtobin if you have no issues with this approach I will submit a pull request.

### PR DESCRIPTION
**Pull request by jacobmerson**
closes #190

### CODE REVIEW

**Comment by wrtobin** [comment]
This looks good to me. 

Offhand @cwsmith do you know if `apf::Mesh::Integrate()` is actually covered by any of the automated test cases?

**Comment by cwsmith** [comment]
Good question.  I don't know. @jacobmerson ?

**Comment by wrtobin** [comment]
Yeah if not this is a decent opportunity to add something basic to cover the existing functionality and the new functionality.

**Comment by jacobmerson** [comment]
I don't think there are any test cases for the integrator currently. Do you have any immediate thoughts on what a decent test case would be?

**Comment by wrtobin** [comment]
We could just load a mesh with a known number of entities and create a counting integrator to count mesh elements and verify the correct number were visited?

**Comment by wrtobin** [comment]
If the integrator only processes owned mesh entities we should be sure to include that consideration in the test case for a simple parallel mesh.

**Comment by jacobmerson** [comment]
Ok. I will think about this tonight and try to put something together.

File: mds/apfMDS.h [hunk 1]
```diff
@@ -53,7 +53,9 @@ typedef std::map<int, MeshEntity*> GlobalToVert;
 Mesh2* makeEmptyMdsMesh(gmi_model* model, int dim, bool isMatched);
 
 /** \brief load an MDS mesh and model from file
-  \param modelfile will be passed to gmi_load to get the model */
+  \param modelfile will be passed to gmi_load to get the model
+  \note gmi_register_mesh and gmi_register_null need to be called before this function
```
Comment (cwsmith) [inline mds/apfMDS.h]: `gmi_register_sim` may also be called to enable loading of GeomSim, Parasolid, and ACIS models

**Comment by jacobmerson** [comment]
@cwsmith I think I address your issue. Is there anything else that needs to be changed before merge?

**Comment by cwsmith** [comment]
@jacobmerson looks good to me.

**Comment by wrtobin** [comment]
@cwsmith I'm gonna merge this in at 4pm today unless anyone has any objections?

**Comment by wrtobin** [comment]
I'm going to merge this now so I don't forget. It is a small enough change that I don't predict any immediate issues resulting, and reverting it in the case an issue does arise should be straightforward.

**Pull request merged**